src/plugins/unified_search/public/saved_query_management/saved_query_management_list.tsx

```diff
@@ -112,41 +166,96 @@ const itemLabel = (attributes: SavedQueryAttributes) => {
+  i18n.translate('unifiedSearch.search.searchBar.savedQueryNoSavedQueriesText', {
+    defaultMessage: 'No saved queries.',
+  }),
+  i18n.translate('unifiedSearch.search.searchBar.savedQueryDescriptionText', {
+    defaultMessage: 'Save query text and filters that you want to use again.',
+  }),
+].join(' ');
+
+const savedQueryMultipleNamespacesDeleteWarning = i18n.translate(
+  'unifiedSearch.search.searchBar.savedQueryMultipleNamespacesDeleteWarning',
+  {
+    defaultMessage: `This saved query is shared in multiple spaces. When you delete it, you remove it from every space it is shared in. You can't undo this action.`,
+  }
+);
+
+const SAVED_QUERY_PAGE_SIZE = 5;
+const SAVED_QUERY_SEARCH_DEBOUNCE = 500;
+
+export const SavedQueryManagementList = ({
   showSaveQuery,
   loadedSavedQuery,
+  savedQueryService,
+  queryBarMenuRef,
   onLoad,
   onClearSavedQuery,
-  savedQueryService,
   onClose,
-  hasFiltersOrQuery,
-}: SavedQueryManagementListProps) {
-  const kibana = useKibana<IUnifiedSearchPluginServices>();
-  const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
+}: SavedQueryManagementListProps) => {
+  const { uiSettings, notifications } = useKibana<IUnifiedSearchPluginServices>().services;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+  const [totalQueryCount, setTotalQueryCount] = useState(0);
+  const [currentPageQueries, setCurrentPageQueries] = useState<SavedQuery[]>([]);
+  const [isInitializing, setIsInitializing] = useState(true);
+  const currentPageFetchId = useRef(0);
+  const selectableRef = useRef<EuiSelectable | null>(null);
   const [selectedSavedQuery, setSelectedSavedQuery] = useState(loadedSavedQuery);
-  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState(null as SavedQuery | null);
+  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState<SavedQuery | null>(null);
   const [showDeletionConfirmationModal, setShowDeletionConfirmationModal] = useState(false);
-  const cancelPendingListingRequest = useRef<() => void>(() => {});
-  const { uiSettings, http, application } = kibana.services;
   const format = uiSettings.get('dateFormat');
 
-  useEffect(() => {
-    const fetchCountAndSavedQueries = async () => {
-      cancelPendingListingRequest.current();
-      let requestGotCancelled = false;
-      cancelPendingListingRequest.current = () => {
-        requestGotCancelled = true;
-      };
+  const debouncedSetSearchTerm = useMemo(() => {
+    return debounce((newSearchTerm: string) => {
+      setSearchTerm((currentSearchTerm) => {
+        if (currentSearchTerm !== newSearchTerm) {
+          setCurrentPageNumber(0);
+        }
 
-      const savedQueryItems = await savedQueryService.getAllSavedQueries();
+        return newSearchTerm;
+      });
+    }, SAVED_QUERY_SEARCH_DEBOUNCE);
+  }, []);
 
-      if (requestGotCancelled) return;
+  const fetchPage = useLatest(async () => {
```

Comment:
Also makes sense to me. Added a loading indicator here, but with a small delay so that it doesn't flicker for fast connections: https://github.com/elastic/kibana/pull/170599/commits/5ea22f5854e5439b90a8e8bda2683bec85538d06.